src/openms/source/ANALYSIS/ID/AhoCorasickDA.cpp

```diff
@@ -0,0 +1,820 @@
+    {
+      build_info_[i].prev = i - 1;
+      build_info_[i].next = i + 1;
+    }
+
+
+    // add new node to bc and set empty head
+    if (empty_head_ == 0)
+    {
+      build_info_[old_size].prev = new_size - 1;
+      build_info_[new_size - 1].next = old_size;
+      empty_head_ = old_size;
+    }
+
+    // add new nodes to empty linkage
+    else
+    {
+      auto emp_tail = build_info_[empty_head_].prev;
+      build_info_[old_size].prev = emp_tail;
+      build_info_[emp_tail].next = old_size;
+      build_info_[empty_head_].prev = new_size - 1;
+      build_info_[new_size - 1].next = empty_head_;
+    }
+  }
+
+  void AhoCorasickDA::buildTrie_(Size begin, const Size& end, const Size& depth, const UInt32& node_pos)
+  {
+    // reaches end of a string
+    if (sequences_[unique_idx_[begin]].size() == depth)
+    {
+      bc_[node_pos].term_flag = 1;
+      output_[node_pos] = unique_idx_[begin];
+      ++begin;
+
+      // string is not substring of another string
+      if (begin == end)
+      {
+        bc_[node_pos].base = 0;
+        bc_[node_pos].leaf_flag = 1;
+        return;
+      }
+    }
+
+    // only one string in subtrie -> store in tail
+    if (begin+1 == end) // && !bc_[node_pos].term_flag)
+    {
+      bc_[node_pos].leaf_flag = 1;
+      bc_[node_pos].base = static_cast<UInt32>(tail_.size());
+      storeInTail_(sequences_[unique_idx_[begin]].substr(depth));
+      output_[-(tail_.size()-2)] = unique_idx_[begin];
+      return;
+    }
+
+    // creates vector containing all transition labels
+    edges_.clear();
+
+    char current_label = sequences_[unique_idx_[begin]][depth];
+
+    for (auto i = begin; i < end; ++i)
+    {
+      char next_label = sequences_[unique_idx_[i]][depth];
+      if (current_label != next_label)
+      {
+        if (isAmbiguous_(current_label))
+        {
+          throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Input peptide must NOT contain ambiguous amino acids (B/J/Z/X) or characters other than amino acids!");
+        }
+        edges_.push_back(current_label);
+        current_label = next_label;
+      }
+    }
+    edges_.push_back(current_label);
+
+
+    // return a base value which is vaild for all labels in edges
+     auto base_value = xCheck_();
+    //auto base_value = plusCheck_();
+
+    // set values
+    bc_[node_pos].base = base_value;
+
+    build_info_[base_value].base_flag = 1;
+
+
+    for (auto label : edges_)
+    {
+      UInt32 child_pos = child_(base_value, label);
+      // need more nodes  //TODO other possibility ?
+      if (child_pos >= bc_.size())
+      {
+        expand_();
+      }
+      arrangeEmpty_(child_pos);
+      bc_[child_pos].lcheck = label;
+      build_info_[child_pos].check = node_pos;
+    }
+
+    // calls buildTrie_ for each subtrie
+    Size begin_sub = begin;
+    auto cur_label = sequences_[unique_idx_[begin]][depth];
+    for (auto end_sub = begin +1; end_sub < end; ++end_sub)
```

Comment:
this iterates over the same column as the loop above right?
The only information needed is the number of sequences with the current AA?
Then why not store this as part of `edges_` (make a vector< AAwithCount>  = vector<pair>) out of it.
Not sure if its faster, but maybe less code.